### ISSUE TITLE
Improve <Select> and use react-aria

### ISSUE THREAD

**Issue by ti10le** [marigold-ui/marigold#1146]
## Description

Our Select component doesn't work like it should. I would use it in #1034 / #1133. This is a good time to make the Select component polymorphic and improve the accessibility (react-aria) like in #1132.

## Context

We want to have a Select component which works correctly and is complete accessible.

## Consequences

Easier and better usage of Select component. Good accesibility!

**Comment by ti10le** [comment]
Rewrite Select component from https://react-spectrum.adobe.com/react-aria/useSelect.html example.
